Openthread apply simplified config (#2645)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_configuring.rst

To use the Thread protocol in |NCS|, set the following Kconfig options:

* :option:`CONFIG_NET_L2_OPENTHREAD` - This option enables the OpenThread stack required for the correct operation of the Thread protocol and allows you to use them.

* General setting options related to network configuration:

  * :option:`CONFIG_NETWORKING`

>   * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread will only work in NCP architecture

Reviewer: Dot missing at the end of the sentence.

>   * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread will only work in NCP architecture.

Reviewer 2: ```suggestion
  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread is meant to work only in the NCP architecture.
```

Reviewer 2: Also, `NCP architecture` can link to the NCP section from https://github.com/nrfconnect/sdk-nrf/pull/2605

Configuring this process is optional, because the Thread :ref:`openthread_samples` in |NCS| use hardcoded network information.

> If you want to manually enable the Thread network Commissioner role on a device, set the following Kconfig option to the provided values:

Reviewer 2: ```suggestion
If you want to manually enable the Thread network Commissioner role on a device, set the following Kconfig option to the provided value:
```

* :option:`CONFIG_OPENTHREAD_COMMISSIONER` to ``y``.

> To enable the Thread network Joiner role on a device, set the following Kconfig option to the provided values:

Reviewer 2: ```suggestion
To enable the Thread network Joiner role on a device, set the following Kconfig option to the provided value:
```
